### ISSUE TITLE
Possible bug on discriminators' indexes

### ISSUE THREAD

**Issue by hyunjinjeong** [Automattic/mongoose#9542]
**Do you want to request a *feature* or report a *bug*?**

Maybe a bug.

**What is the current behavior?**

When defining a `unique index` on discriminators, it is created as a `partial unique index` in MongoDB. However, if I define a `non-unique index` on discriminators, it is not created as a `partial index` in MongoDB.

**If the current behavior is a bug, please provide the steps to reproduce.**

I made [a gist](https://gist.github.com/hyunjinjeong/cd1838ea0683046b3f1d1c324c24dfb5) for the model code.

Below is where I define indexes on the discriminators.

![defining_indexes](https://user-images.githubusercontent.com/11304601/99058325-6443d080-25e0-11eb-9aeb-cececd45a5e7.png)

Below is the result when I `getIndexes()` on MongoDB.

![result_get_indexes](https://user-images.githubusercontent.com/11304601/99058700-eaf8ad80-25e0-11eb-977f-7726df6cea7c.png)

As shown in the screenshot, the `placeId` index is a `unique partial index`, but the `userId` index is a normal index.

**What is the expected behavior?**

Indexes on discriminators should be created as `partial indexes` in MongoDB.

I am not sure if it is a bug or an intended implementation, but I have not found related issues in this repo so I report this issue. 

**What are the versions of Node.js, Mongoose and MongoDB you are using? Note that "latest" is not a version.**

- Node.js: v14.15.0
- Mongoose: v5.10.14
- MongoDB: v4.4.1

**Comment by hyunjinjeong** [comment]
Or is using `partialFilterExpression` directly the only way to make indexes on discriminators?

```typescript
BookmarkBasisSchema.index(
  {
    userId: 1,
  },
  {
    partialFilterExpression: {
      type: 'USER',
    },
  }
)
```

The index was set without any issues when using it.

![use_partial_filter_expression](https://user-images.githubusercontent.com/11304601/99086361-c0244e80-260c-11eb-8c89-0f794350ae37.png)

**Comment by vkarpov15** [comment]
We started setting `partialFilterExpression` automatically for `unique` indexes in v5.2.0 with #6347. I think we should be good to make this fix for v5.10.15.

As a workaround, you can set `partialFilterExpression` yourself as you did in https://github.com/Automattic/mongoose/issues/9542#issuecomment-726817254 :+1:

**Comment by hyunjinjeong** [comment]
I appreciate your work. Thank you!
